accord-core/src/test/java/accord/burn/BurnTest.java

```diff
@@ -551,40 +549,22 @@ public void testOne()
     private static void run(long seed)
     {
         Duration timeout = Duration.ofMinutes(3);
-        Runnable fn = () -> run(seed, 1000);
-        AsyncResult.Settable<?> promise = AsyncResults.settable();
-        Thread t = new Thread(() -> {
-            try
-            {
-                fn.run();
-                promise.setSuccess(null);
-            }
-            catch (Throwable e)
-            {
-                promise.setFailure(e);
-            }
-        });
-        t.setName("BurnTest with timeout");
-        t.setDaemon(true);
         try
         {
-            t.start();
-            AsyncChains.getBlocking(promise, timeout.toNanos(), TimeUnit.NANOSECONDS);
+            TimeoutUtils.runBlocking(timeout, "BurnTest with timeout", () -> run(seed, 1000));
         }
         catch (Throwable thrown)
         {
             Throwable cause = thrown;
             if (cause instanceof ExecutionException)
                 cause = cause.getCause();
-            if (cause instanceof InterruptedException || cause instanceof TimeoutException)
-                t.interrupt();
             if (cause instanceof TimeoutException)
             {
                 TimeoutException override = new TimeoutException("test did not complete within " + timeout);
                 override.setStackTrace(new StackTraceElement[0]);
                 cause = override;
             }
-            logger.error("Exception running burn test for seed {}:", seed, t);
+            logger.error("Exception running burn test for seed {}:", seed, cause);
```

Comment:
`t` was a bug, ment to do `cause` or `thrown` to see what actually caused the failure